### ISSUE TITLE
SQLite graph creation issue

### ISSUE THREAD

**Issue by sualeh** [sualeh/SchemaCrawler#112]
_From @maxg2k on February 14, 2017 7:2_

Hi I'm getting the following error.  Any idea?

Feb 14, 2017 6:44:09 AM us.fatehi.commandlineparser.CommandLineUtility logSafeArguments
INFO: SchemaCrawler, v14.14.01
Feb 14, 2017 6:44:09 AM us.fatehi.commandlineparser.CommandLineUtility logSafeArguments
INFO: Command line:
-server=sqlite
-user=
-password=*****
-database=/share/tasks.db
-tabletypes=TABLE
-infolevel=maximum
-routines=
-command=graph
-outputfile=/tmp/tasks_db.png
-loglevel=CONFIG
Feb 14, 2017 6:44:09 AM us.fatehi.commandlineparser.CommandLineUtility logFullStackTrace
SEVERE: Cannot perform operation: [SQLITE_ERROR] SQL error or missing database (near "graph": syntax error)
schemacrawler.schemacrawler.SchemaCrawlerException: Cannot perform operation: [SQLITE_ERROR] SQL error or missing database (near "graph": syntax error)
        at schemacrawler.tools.text.operation.OperationExecutable.executeOn(OperationExecutable.java:129)
        at schemacrawler.tools.executable.SchemaCrawlerExecutable.executeOn(SchemaCrawlerExecutable.java:120)
        at schemacrawler.tools.executable.BaseStagedExecutable.execute(BaseStagedExecutable.java:91)
        at schemacrawler.tools.commandline.SchemaCrawlerCommandLine.execute(SchemaCrawlerCommandLine.java:129)
        at schemacrawler.Main.main(Main.java:90)
**Caused by: java.sql.SQLException: [SQLITE_ERROR] SQL error or missing database (near "graph": syntax error)**
        at org.sqlite.DB.newSQLException(DB.java:888)
        at org.sqlite.DB.newSQLException(DB.java:899)
        at org.sqlite.DB.throwex(DB.java:866)
        at org.sqlite.NativeDB.prepare(Native Method)
        at org.sqlite.DB.prepare(DB.java:208)
        at org.sqlite.Stmt.execute(Stmt.java:144)
        at sf.util.DatabaseUtility.executeSql(DatabaseUtility.java:145)
        at schemacrawler.tools.text.operation.OperationExecutable.executeOn(OperationExecutable.java:118)
        ... 4 more

_Copied from original issue: sualeh/SchemaCrawler-Docker#1_

**Comment by sualeh** [comment]
@maxg2k I have not seen this issue before - please could you email `/share/tasks.db` to me at sualeh@hotmail.com?

**Comment by sualeh** [comment]
Max,

[The **graph** command was removed in 14.14.01](http://sualeh.github.io/SchemaCrawler/changes-report.html). Please re-run with the following command-line, and let me know if it works:

```
schemacrawler.sh -server=sqlite -database=/share/tasks.db -infolevel=maximum -command=details -routines= -tabletypes=TABLE -outputformat=pdf
```

Sualeh.

**Comment by maxg2k** [comment]
I did try the PDF command, albeit with user and password blanks provided inline, as below. But no luck.  The PDF files got generated but they are blank... (find attached)

Command I ran:
./schemacrawler.sh -server=sqlite -user= -password= -database=/share/tasks.db -infolevel=maximum -command=details -routines= -tabletypes=TABLE -outputformat=pdf
[sc.8rks8k3v.pdf](https://github.com/sualeh/SchemaCrawler/files/776082/sc.8rks8k3v.pdf)
[sc.kpsmggr8.pdf](https://github.com/sualeh/SchemaCrawler/files/776083/sc.kpsmggr8.pdf)

**Comment by maxg2k** [comment]
tried another command without -outputformat=pdf switch and got the details into a text file successfully.

**./schemacrawler.sh -server=sqlite -user= -password= -database=/share/tasks.db -infolevel=maximum -command=details -routines= -tabletypes=TABLE**

But I'm looking for ER Diagram generation using GraphViz...  So that's no longer possible with v14.14.01 ?

**Comment by sualeh** [comment]
I can see the E-R diagram with all of the tables in your PDF files. What PDF file viewer are you using?

**Comment by sualeh** [comment]
Please make sure that you have read the [SchemaCrawler diagramming](http://sualeh.github.io/SchemaCrawler/diagramming.html) notes.

**Comment by mambax7** [comment]
same issue here with version 14.21.02 on Windows 10
I'm using exactly the same command as in the diagram example, except I replaced pdf with png, but pdf was the same:

> ..\..\_schemacrawler\schemacrawler.cmd -server=mysql -host=localhost -port=3307 -database=260xx -user=root -password=  -infolevel=maximum -command=schema -outputformat=png -outputfile=database-diagram.png

It outputs only list of tables and not a graph:
![image](https://user-images.githubusercontent.com/613686/43039494-6c00db1c-8cfc-11e8-996a-839974d72965.png)

**Comment by sualeh** [comment]
Please attach the logs, @mambax7 - you can get them with -loglevel=ALL

**Comment by mambax7** [comment]
attached:
[log.zip](https://github.com/schemacrawler/SchemaCrawler/files/2216462/log.zip)

**Comment by sualeh** [comment]
@mambax7 - please run the following query on your database, and see if it returns any data

```
SELECT 
	* 
FROM 
	INFORMATION_SCHEMA.TABLE_CONSTRAINTS 
WHERE 
	INFORMATION_SCHEMA.TABLE_CONSTRAINTS.CONSTRAINT_TYPE = 'FOREIGN KEY' 
```

**Comment by mambax7** [comment]
no, it doesn't. 
I thought, it will try to guess potential foreign keys.

**Comment by mambax7** [comment]
OK, I've installed the SakilaDB, and the PDF is working now (using Foxit reader). 
But the PNG  created a file with 0 size.  SVG created a file, but when I tried to see it in the browser, it was empty. 

However, the issue is that SchemaCrawler printed in the PDF "all" databases present server, even if I specifically set only the Sakila DB. 
This has caused issues on my other server, when one of the databases had a corrupt table, because - SchemaCrawler just stopped:

> Error: Could not retrieve columns for table <257final.x449_extcal_cat>: Table '257final.x449_extcal_cat' doesn't exist
> 
> Query being executed when exception was thrown:
> SHOW FULL COLUMNS FROM `x449_extcal_cat` FROM `257final` LIKE '%'

Why do we have to specify "-database=" if it is being ignored? 

Otherwise, awesome tool!!!
